doc: Create pull_request_template.md

jcsp opened the pull request:
Let's take the opportunity to nudge new contributors
to have well formatted commit messages.

Signed-off-by: John Spray <john.spray@redhat.com>

batrick commented:
I'd suggest also telling them that the ticket is especially important if they want their fix backported as that creates an incentive.

jcsp commented:
Added the note about creating tickets for bugs, and moved into .github.

@tchaikov the HTML style comment is what I've seen in other examples -- it seems to work, not sure if it's a github specific hack for templates, or if it just works because the html comment marks get passed through.

gregsfortytwo commented:
We shouldn't ask for a Signed-off-by in official documentation without linking to a statement about what it actually means. :) Rest looks good!

LenzGr commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,39 @@
+small or obvious fixes, the tracker ticket is important to enable
+the backporting process to stable branches.
+
+More information for contributors is available here:
```
I would move this reference further up, directly after the "Thank you!" line. Then I would start the following part with "In a nutshell, this is what you should do:"

tchaikov commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,39 @@
+
+-->
+
+- [] References tracker ticket
```
i think it's needed.

smithfarm replied:
yes, the space is needed - verified just now

jcsp commented:
Updated to call out Signed-off-by's importance, add the space to the `[ ]` bits, and edit it down a bit.

@LenzGr I'm avoiding putting the SubmittingPatches.rst reference right at the top, because I want peoples eyeballs to gravitate to that example commit message -- once someone goes to SubmittingPatches I worry they get lost in the details, so I'm leaving that link til the end.
